server/src/main/java/io/crate/operation/collect/files/CSVLineParser.java

```diff
@@ -33,16 +33,32 @@
 import java.io.IOException;
 import java.nio.charset.StandardCharsets;
 import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
 
 public class CSVLineParser {
 
-    private final ArrayList<String> keyList = new ArrayList<>();
     private final ByteArrayOutputStream out = new ByteArrayOutputStream();
+    private final Map<Integer, String> keyPositions = new HashMap<>();
+    private final ArrayList<String> headerKeyList = new ArrayList<>();
```

Comment:
Wouldn't it be possible to remove the headerKeyList, and use the keys of `keysPositions` to do the same checks?

Reply:
I think not with the current logic of the implementation. If we want to keep the exception about consistent csv:
`throw new IllegalArgumentException("Number of values exceeds number of keys in csv file");`
All csv header parts are kept separately from csv header parts that are part of the target columns.